tei_adapter/tei_id_extractor/src/main/scala/weco/catalogue/tei/id_extractor/TeiIdExtractorWorkerService.scala

```diff
@@ -48,33 +46,54 @@ class TeiIdExtractorWorkerService[Dest](
     Flow[(SQSMessage, NotificationMessage)]
       .map {
         case (msg, NotificationMessage(body)) =>
-          (Context(msg), fromJson[TeiFileChangedMessage](body).toEither)
+          (Context(msg), fromJson[TeiPathMessage](body).toEither)
       }
       .via(catchErrors)
 
-  def filterNonTei = Flow[(Context, TeiFileChangedMessage)].filter {
-    case (_, TeiFileChangedMessage(path, _, _)) => !isTeiFile(path)
+  def filterNonTei = Flow[(Context, TeiPathMessage)].filter {
+    case (_, teiPathMessage) => !isTeiFile(teiPathMessage.path)
   }
 
   def processMessage =
-    Flow[(Context, TeiFileChangedMessage)]
+    Flow[(Context, TeiPathMessage)]
       .filter {
-        case (_, TeiFileChangedMessage(path, _, _)) => isTeiFile(path)
+        case (_, teiPathMessage) => isTeiFile(teiPathMessage.path)
+      }
+      .via(broadcastAndMerge(processDeleted, processChange))
+
+  def processDeleted =
+    Flow[(Context, TeiPathMessage)]
+      .collect {
+        case (ctx, msg) if msg.isInstanceOf[TeiPathDeletedMessage] =>
+          (ctx, msg.asInstanceOf[TeiPathDeletedMessage])
+      }
+      .delay(config.deleteMessageDelay)
```

Comment:
Is this a normal thing, or is there a reason we delay deleting messages here?

Reply:
yeah, there should be a comment here, I will add. The reason is that if something is moved in the repo, the updater lambda has no way that is the same thing moved, cause you only know exactly what it is when you inspect the XML and get the id and the lambda never sees the XML. So what happens in that case is that the lambda sends a deleted message for the old path and a changed message for the new path. If the delete message is processed before the change message we may send a deleted message downstream. It's not a massive issue cause the change message will arrive after and override the delete one but it may cause a 'blip' where something disappears and then reappears. The delay is to avoid that 'blip'